### PR TITLE
JetBrains: Remove usage of APIs added after 2021.1

### LINKED ISSUE #37475: JetBrains: Audit 2020.2 functionality
While debugging an issue I tried starting 2020.2 and found _numerous_ APIs that we use which aren't supported. Some of them are easy to fix, some of them required more work:

- We're using JCEF APIs that were added later but we can use deprecated APIs to work around the issue.
- The link implementation added for the code preview does not work.
- Maybe more areas? I did not get it to compile yet.

### PR DESCRIPTION
**Pull request by philipp-spiess**
Closes #37475 

This still needs testing. I was not able to run on macOS versions prior to 2022.1 because of the JCEF bundling issue we were seeing at the very beginning of our trial. Will verify on Windows instead.

## The changes:

- JBCefBrowserBase → JBCefBrowser
- AnActionLink → LinkLabel
- SourcegraphJBCefBrowser#setPageBackgroundColor → removed

## Current support

With these changes, I was able to verify support for these versions:

```
IC-2022.1.3 ✅
IC-2021.3.3 ✅
IC-2021.2.4 ✅
IC-2021.1.3 ✅ 
IC-2020.3.4 ❌
IC-2020.2.4 ❌
```

`2020.2` and `2020.3` are limited by [this issue](https://github.com/sourcegraph/sourcegraph/issues/37475#issuecomment-1171355831).

## Decision:

We decided to drop support for 2020.2 and 2020.3

## Test plan

I ran all of these versions on our Debian test machine. Here's an example of a working version (2021.1):

Screenshot 2022-06-30 at 17 27 09
Screenshot 2022-06-30 at 17 26 49

```
α jetbrains (ps/jetbrains-2020.2-support) ./gradlew listProductsReleases
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :listProductsReleases
IC-2022.1.3
IC-2021.3.3
IC-2021.2.4
IC-2021.1.3
BUILD SUCCESSFUL in 6s
1 actionable task: 1 executed
```

## App preview:

- [Web](https://sg-web-ps-jetbrains-2020-2-support.onrender.com)
- [Storybook](https://5f0f381c0e50750022dc6bf7-bwyugxljxa.chromatic.com)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.

### CODE REVIEW

File: client/jetbrains/src/main/java/com/sourcegraph/browser/SourcegraphJBCefBrowser.java [hunk 1]
```diff
@@ -14,7 +14,6 @@ public SourcegraphJBCefBrowser(@NotNull JSToJavaBridgeRequestHandler requestHand
         super("http://sourcegraph/html/index.html");
         // Create and set up JCEF browser
         CefApp.getInstance().registerSchemeHandlerFactory("http", "sourcegraph", new HttpSchemeHandlerFactory());
-        this.setPageBackgroundColor(ThemeUtil.getPanelBackgroundColorHexString());
```
Comment (philipp-spiess) [inline client/jetbrains/src/main/java/com/sourcegraph/browser/SourcegraphJBCefBrowser.java:17]: Verified that this is no longer needed as we manually control when to show the web view now. The method for it was added after 2020.2

Comment (vdavid) [reply client/jetbrains/src/main/java/com/sourcegraph/browser/SourcegraphJBCefBrowser.java:17]: Great find!

**Comment by vdavid** [comment]
The changes I found were:
- JBCefBrowserBase → JBCefBrowser
- AnActionLink → LinkLabel
- SourcegraphJBCefBrowser#setPageBackgroundColor → removed
All three look good to me. Do you need any help with the testing on any OS / IDE version?

**Comment by philipp-spiess** [comment]
Apparently it's the custom headers in the GQL request 🤔 🤔 

Screenshot 2022-06-30 at 15 20 32

**Comment by philipp-spiess** [comment]
I bumped the min version to 2021.1:

```
α jetbrains (ps/jetbrains-2020.2-support) ./gradlew listProductsReleases
Starting a Gradle Daemon (subsequent builds will be faster)

> Task :listProductsReleases
IC-2022.1.3
IC-2021.3.3
IC-2021.2.4
IC-2021.1.3
BUILD SUCCESSFUL in 6s
1 actionable task: 1 executed
```

**Pull request merged**